jobs/shield-agent/templates/bin/post-start

```diff
@@ -4,6 +4,19 @@ export PATH=${PATH}:/var/vcap/packages/shield/bin
 export PATH=${PATH}:/var/vcap/packages/utils/bin
 SHIELD_CONFIG=$(mktemp --tmpdir auto.XXXXXX)
 trap "rm -rf ${SHIELD_CONFIG}" INT TERM QUIT EXIT
+<%
+def format_name(name, err_ctx)
+   var = name.gsub("<deployment_name>", spec.deployment)
+   var = var.gsub("<instance_index>", spec.index.to_s)
+   var = var.gsub("<ip>", spec.ip)
+   var = var.gsub("<instance_name>", spec.name)
+   var = var.gsub("<instance_zone>", spec.az)
```

Comment:
How do you feel about `(...)` delimiters, instead of `<...>`?

Reply:
I definitely see no objection at all, but let me first explain how we ended-up with this choice.

The thing is that the GUI doesn't escape any angle brackets from SHIELD resource names. So, in the current version when you put those angle brackets, they get interpreted as unknown HTML tags by the web browser and they end up not being displayed.

When we saw that, we said it could be a good choice to use them, because nobody could ever have angle brackets in their SHIELD resources names, due to that bug. On the opposite, we were worried that anyone having parenthesis in such names could suddenly get an error after upgrading SHIELD, because they would then be interpreted as a templating syntax.

Now with this explanation, do you still prefer parenthesis over angle brackets?

Reply:
I'd rather not rely on input sanitation bugs in the web UI, esp. with v8's new UI coming down the pipe.

The templating system would only encroach on existing names with parentheses if those parentheses are surrounding new keywords, right?  I.e. "My New (Old) System" would be fine, but "My other (deployment) Thing" wouldn't?

If it is a worry still, perhaps we should use a (new) subkey like `template` that supersedes the name key.

Reply:
There was prior art for this, which I have been unable to find.  It was either toolbelt or jumpbox or maybe vault, but it used keywords like: `(deployment)`, `(name)`, `(az)`, `(id)`, and `(index)`, and it worked beautifully.

Reply:
Ok, let's go for `(deployment)`, `(index)`, `(ip)`, `(name)`, and `(az)`.

We initially thought we would raise an error in case we face an unknown tag (this is the current implementation), but as I read you I think it would be better just to leave those alone unchanged.

So, let's remove the error `raise`d when facing something like `My New (Old) System`.